shared/router-v2/router.shared.js

```diff
@@ -86,19 +99,27 @@ export const oldActionToNewActions = (action: any, navigation: any) => {
         ? [switchStack, StackActions.reset({actions: sa, index: sa.length - 1})]
         : [switchStack]
     }
-    case RouteTreeGen.clearModals:
-      const path = Constants._getModalStackForNavigator(navigation.state)
-      let numModals = 0
-      path.reverse().some(p => {
-        if (modalRoutes[p.routeName]) {
-          numModals++
-          return false
-        }
-        return true
-      })
-
+    case RouteTreeGen.clearModals: {
+      const numModals = getNumModals(navigation)
       return numModals ? [StackActions.pop({n: numModals})] : []
+    }
     case RouteTreeGen.navigateUp:
       return [StackActions.pop()]
+    case RouteTreeGen.popToScreen: {
+      const modals = getNumModals(navigation)
+      const appPath = Constants._getAppStackForNavigator(navigation.state)
+      if (!appPath.length) {
```

Comment:
unclear if this is what we want. seems like it should just go back through everything. if it finds it it pops to there.aka if my 'target' is the root modal this'll do nothing (etc)

Reply:
Good point, yeah this assumes the target is in the app path.

Reply:
@chrisnojima so there was another problem with this. If we're on the infopanel modal, we can't `pop` past the conversation, because that is in a different navigator. I changed it to use `NavigationActions.navigate`.